### PR TITLE
Disable where ops in xla auto-clustering

### PR DESCRIPTION
**Pull request by wenscarl**
This PR mainly aims at excluding where op in XLA auto-clustering. A possible issue caused by where being included in auto-clustering is [here](https://github.com/tensorflow/tensorflow/issues/55610). A TF_XLA_FLAGS, called `tf_xla_cluster_exclude_ops` is added, which can only take string values from `Where`, and some_others_ops in the future(separate by comma). By default, where op is included from auto-clustering. The way to disable it from auto-clustering looks like,
`TF_XLA_FLAGS=--tf_xla_cluster_exclude_ops=Where` .

### CODE REVIEW

**Comment by nluehr** [comment]
CC @jurahul

**Review by smit-hinsu — APPROVED** [review]

**Comment by cheshire** [comment]
Do you think we could do something much simpler, without a flag, and just disabling tf.where?

**Comment by wenscarl** [comment]
> Do you think we could do something much simpler, without a flag, and just disabling tf.where?

The reasons I made the disabling of tf.where with a flag are:
1) Make the change scalable and extensible to other ops if perf regression is spotted in the future. Adding too many env vars(per op) doesn't look clean. 

2) The use may have a option to turn off or on. I don't think tf.where in auto-clustering will always hurt.

**Review by cheshire — CHANGES_REQUESTED** [review]
Maybe add a test that tf.where is not actually autoclustered?

**Comment by wenscarl** [comment]
> Forking a process and writing to a global directory for a test is really not great. Maybe look at existing C++ tests for autoclustering?
Could you give me a pointer where to add the C++ test? How about `compilability_check_util_test.cc` ?

**Comment by cheshire** [comment]
I would expect something along the lines of https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/compiler/jit/mark_for_compilation_pass_test.cc;l=180-197

**Comment by wenscarl** [comment]
> I would expect something along the lines of https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/compiler/jit/mark_for_compilation_pass_test.cc;l=180-197

Test fail with
`bazel test -test_env=TF_XLA_FLAGS=--tf_xla_cluster_exclude_ops=Where   --config=cuda  -c opt //tensorflow/compiler/jit:compilation_passes_test 
`
and pass with 
`bazel test  --config=cuda  -c opt //tensorflow/compiler/jit:compilation_passes_test 
`
which are as expected.

File: tensorflow/compiler/jit/mark_for_compilation_pass_test.cc [hunk 1]
```diff
@@ -196,6 +196,25 @@ TEST(XlaCompilationTest, StringUnsupported) {
   EXPECT_TRUE(clusters.empty());
 }
 
+TEST(XlaCompilationTest, WhereUnsupported) {
```
Comment (wenscarl) [inline tensorflow/compiler/jit/mark_for_compilation_pass_test.cc:199]: This test was added.

**Review by cheshire — APPROVED** [review]

**Pull request merged**